flytekit/core/condition.py

```diff
@@ -35,27 +35,30 @@ def name(self):
 
     # Output Node or None
     def __call__(self, **kwargs):
-        # Eval every ConjunctionExpression and ComparisonExpression
-        self._cs.eval_by_kwargs(**kwargs)
         for c in self._cs.cases:
-            # c._expr = _update_promise(c.expr, kwargs)
-
+            c._expr = _update_promise(c.expr, kwargs)
```

Comment:
This line directly change the expression of conditional section. Would cause error when we call Branchnode Twice. I provide a little preseudo code on when it would failed.


```
@workflow
def branching(x: int) -> int:
    return conditional("test").if_(x == 2).then(t()).else_().then(wf1())

@workflow
def call_branching_twice(x: int, y:int) -> Tuple[int, int]:
    return branching(x=x), branching(x=y) # This would cause error because expression is overwrited
```

Reply:
will take a look